src/libraries/Native/Unix/System.Native/pal_iosversion.m

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_iosversion.h"
+#import <Foundation/Foundation.h>
+
+const char* SystemNative_iOSSupportVersion()
+{
+    @autoreleasepool
```

Comment:
I am not necessarily sure whether it's needed or not but LLVM code has it there (https://github.com/llvm/llvm-project/blob/62ec4ac90738a5f2d209ed28c822223e58aaaeb7/lldb/source/Host/macosx/objcxx/HostInfoMacOSX.mm#L83-L90).

Reply:
yeah I saw that, I was mostly wondering because we have some custom code in pal_autoreleasepool.m that does thread initialization stuff used by ThreadPool code in managed and I don't know if we need that here too:

https://github.com/dotnet/runtime/blob/5b683e2076f35e74d550ffa4a3182ebd6726b3c8/src/libraries/Native/Unix/System.Native/pal_autoreleasepool.m#L18-L35

@rolfbjarne do you know the implications?

Reply:
Good point. I'm no export on that but the documentation really suggest that may be necessary... 🤷🏻‍♂️

Reply:
That's fine [1] if the new API are called from user+managed code.

If the dotnet runtime(s) calls (independently or before xamarin's runtime completed it's initialization) `SystemNative_iOSSupportVersion` then `SystemNative_CreateAutoreleasePool` should be called first (or to create the instance).

[1] if not it means we need to fix it

Reply:
I'll modify it to call `SystemNative_CreateAutoreleasePool` instead of `@autoreleasepool` which should be safe. The code should not depend on Xamarin runtime being used/initialized (even if in most cases it will be).

Reply:
The Apple documentation says that using `NSAutoreleasePool` manually would trip the ARC analyzer so I just extracted the code into helper method and kept using `@autoreleasepool`.